client/pkg/omni/resources/infra/machine_registration.go

```diff
@@ -0,0 +1,45 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package infra
+
+import (
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+
+	"github.com/siderolabs/omni/client/pkg/omni/resources"
+	"github.com/siderolabs/omni/client/pkg/omni/resources/system"
+)
+
+// NewMachineRegistration creates a new MachineRegistration resource.
+func NewMachineRegistration(id string) *MachineRegistration {
```

Comment:
Is this type only for the static infra providers?

Reply:
yep

Reply:
It's actually for any providers